carmin-server/server/test/resources/test_executions.py

```diff
@@ -0,0 +1,101 @@
+from server.test.fakedata.executions import (
+    POST_VALID_EXECUTION, POST_INVALID_EXECUTION_FILE_NOT_EXIST,
+    POST_INVALID_EXECUTION_ARRAY_FILE_NOT_EXIST, POST_INVALID_IDENTIFIER_SET,
+    POST_INVALID_EXECUTION_IDENTIFIER_NOT_EXIST, POST_INVALID_MODEL)
+from server.test.fakedata.users import standard_user
+from server.test.utils import get_test_config, error_from_response
+
+
+@pytest.yield_fixture()
+def test_config(tmpdir_factory):
+    test_config = get_test_config()
+    test_config.db.session.add(standard_user())
+    test_config.db.session.commit()
+
+    pipelines_root = tmpdir_factory.mktemp('pipelines')
+    data_root = tmpdir_factory.mktemp('data')
+    pipelines_root.join('pipeline1.json').write(
+        json.dumps(PipelineSchema().dump(PIPELINE_FOUR).data))
+    user_dir = data_root.mkdir(standard_user().username)
+    user_dir.join('test.txt').write('test file')
+    user_execution_dir = user_dir.mkdir('executions')
+    app.config['DATA_DIRECTORY'] = str(data_root)
+    app.config['PIPELINE_DIRECTORY'] = str(pipelines_root)
+
+    yield test_config
+
+    test_config.db.drop_all()
+
+
+class TestExecutionsResource():
+    # tests for POST
+    def test_post_valid_execution(self, test_config):
+        user_execution_dir = os.path.join(app.config['DATA_DIRECTORY'],
+                                          standard_user().username,
+                                          'executions')
+        response = test_config.test_client.post(
+            '/executions',
+            headers={"apiKey": standard_user().api_key},
+            data=json.dumps(ExecutionSchema().dump(POST_VALID_EXECUTION).data))
+        assert os.listdir(user_execution_dir)
+        assert response.status_code == 200
+
+    def test_post_file_doesnt_exist(self, test_config):
+        user_execution_dir = os.path.join(app.config['DATA_DIRECTORY'],
+                                          standard_user().username,
+                                          'executions')
+        response = test_config.test_client.post(
+            '/executions',
+            headers={"apiKey": standard_user().api_key},
+            data=json.dumps(ExecutionSchema().dump(
+                POST_INVALID_EXECUTION_FILE_NOT_EXIST).data))
+        assert not os.listdir(user_execution_dir)
+        assert response.status_code == 400
+
+    def test_post_array_file_doesnt_exist(self, test_config):
+        user_execution_dir = os.path.join(app.config['DATA_DIRECTORY'],
+                                          standard_user().username,
+                                          'executions')
+        response = test_config.test_client.post(
+            '/executions',
+            headers={"apiKey": standard_user().api_key},
+            data=json.dumps(ExecutionSchema().dump(
+                POST_INVALID_EXECUTION_ARRAY_FILE_NOT_EXIST).data))
+        assert not os.listdir(user_execution_dir)
+        assert response.status_code == 400
```

Comment:
If you are expecting a given ErrorAndMessage, please validate that the response body is this ErrorCodeAndMessage (refer to /authenticate and /register for example).

Reply:
... or just like you do in the next test. (Please change this for all your tests)